play.pokemonshowdown.com/src/client-connection.ts

```diff
@@ -13,70 +13,173 @@ declare const POKEMON_SHOWDOWN_TESTCLIENT_KEY: string | undefined;
 export class PSConnection {
 	socket: any = null;
 	connected = false;
-	queue = [] as string[];
+	queue: string[] = [];
+	private reconnectDelay = 1000;
+	private reconnectCap = 15000;
+	private shouldReconnect = true;
```

Comment:
`shouldReconnect` should probably be a function checking if uptime is under a day.